ultravox-client/ultravox_client/session.py

```diff
@@ -291,11 +291,16 @@ async def send_text(self, text: str):
 
     async def send_data(self, msg: dict):
         """Sends a data message to the Ultravox server. See https://docs.ultravox.ai/api/data_messages."""
-        if not self._room:
+        if not self._room or not self._socket:
             raise RuntimeError("Cannot send data while not connected")
         if "type" not in msg:
             raise ValueError("Message must have a 'type' field")
-        await self._room.local_participant.publish_data(json.dumps(msg).encode("utf-8"))
+        if msg["type"] == "client_tool_result":
+            await self._socket.send(json.dumps(msg))
+        else:
+            await self._room.local_participant.publish_data(
+                json.dumps(msg).encode("utf-8")
+            )
```

Comment:
Given that it's just as easy to check the length (and we know that big messages are likely to get dropped) I'd be inclined to change the condition to `len(jsonified) > 512` (or 1000 or whatever)?

Reply:
Chatted offline. I've come around because:

* The non-essential message types I'd like to keep from clogging up this channel also happen to be small anyway.
* This avoids an implicit limit on sendText.
* This is future-proof in that the SDK can handle new large messages added after its publication also.